adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -41,105 +39,75 @@ sealed abstract class ShuffleRegionJoin[T: ClassTag, U: ClassTag, RT, RU]
   protected val sd: SequenceDictionary
   protected val partitionSize: Long
   protected val sc: SparkContext
+  protected val partitionMap: Seq[Option[(ReferenceRegion, ReferenceRegion)]]
 
   // Create the set of bins across the genome for parallel processing
-  //   partitionSize (in nucleotides) may range from 10000 to 10000000+ 
+  //   partitionSize (in nucleotides) may range from 10000 to 10000000+
   //   depending on cluster and dataset size
   protected val seqLengths = Map(sd.records.map(rec => (rec.name, rec.length)): _*)
   protected val bins = sc.broadcast(GenomeBins(partitionSize, seqLengths))
 
   /**
-   * Performs a region join between two RDDs (shuffle join).
+   * Performs a region join between two RDDs (shuffle join). All data should be pre-shuffled and
```

Comment:
This might be difficult because the extension of RegionJoin requires this signature.